app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,76 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all.order('created_at DESC')
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
+    set_post
+  end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
+
+  # GET /posts/1/edit
+  def edit
+    set_post
```

Comment:
You can set a before_ action to handle this.in case of any ambiguity don't hesitate to contact me.